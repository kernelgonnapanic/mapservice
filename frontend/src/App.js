import React from 'react'
import Navigation from './components/Navigation'
import AddPlace from './components/AddPlace'
import { store } from './components/redux/reducers'
import { Provider } from 'react-redux'

const App = () => {
	return (
		<Provider store={store}>
			<Navigation />
		</Provider>
	)
}

export default App
