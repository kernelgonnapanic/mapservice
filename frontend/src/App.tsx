import React from 'react'
import { Provider } from 'react-redux'
import { createGlobalStyle, ThemeProvider } from 'styled-components'
import Navigation from './components/Navigation/Navigation'
import { store } from './components/redux/reducers'


const theme = {
	colors: {
		mainDark: '#203A55',
		mediumGray: '#546577',
		white: '#FFFFFF',
		purple: 'rgb(209, 196, 233)',
		gray: 'hsl(220, 7%, 83%)',
	},
	headerHeight: '64px',
	barHeight: '75px',
	boxshadow: '0 3px 0 hsl(220, 7%, 83%) ',
	borderRadius: {
		small: 5,
		medium: "20px"
	},
	fontSize: {
		s: "14px",
		l: "16px",
		xl: "20px",
		xxl: "30px"
	}
}

const GlobalStyle = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css?family=Varela+Round&display=swap');


  html, body, #root {
 	 height: 100%
	}

  body {
	margin: 0;
	padding: 0;
	background-color: #FBFBFB;
	font-family: 'Varela Round', sans-serif;
  }

  *, *::before, *::after {
	  box-sizing: border-box
  }
`

const App: React.FC = () => {

	return (
		<Provider store={store}>
			<GlobalStyle />
			<ThemeProvider theme={theme}>
				<Navigation />
			</ThemeProvider>
		</Provider>
	)
}

export default App
