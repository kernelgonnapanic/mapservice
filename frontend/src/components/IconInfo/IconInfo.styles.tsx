import styled from "styled-components";

export const Text = styled.span`
    display: flex;
    align-items: center;
    margin-left: 25px;
    color: ${({theme}) => theme.colors.gray}
`

export const Circle = styled.div`
     display: flex;
     color: ${({theme}) => theme.colors.white}};
     justify-content: center;
     align-items: center;
     width: 35px;
     height: 35px;
     border-radius: 25px;
     background-color: ${({theme}) => theme.colors.lightGray}    
`

export const Wrapper = styled.div`
    display: flex;
    margin: 10px
    
`