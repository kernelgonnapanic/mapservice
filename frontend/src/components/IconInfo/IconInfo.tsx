import React from 'react'
import { Text, Circle, Wrapper } from './IconInfo.styles'
import { Search } from 'react-feather'

interface Props {
	icon: JSX.Element
	text?: string | number | undefined
}

const IconInfo: React.FC<Props> = ({ icon, text }) => {
	return (
		<Wrapper>
			<Circle>{icon}</Circle>
			<Text>{text}</Text>
		</Wrapper>
	)
}

export default IconInfo
