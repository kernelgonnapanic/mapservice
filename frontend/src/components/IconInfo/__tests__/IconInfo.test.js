import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../../App'
import IconInfo from '../IconInfo'
import React from 'react'
import { Search } from 'react-feather'
import ReactDOM from 'react-dom'

test('render IconInfo component', async () => {
	const { asFragment } = render(
		<ThemeProvider theme={theme}>
			<IconInfo text="Hello!" />
		</ThemeProvider>,
	)
	expect(asFragment()).toMatchSnapshot()
})

test('render IconInfo text with prop', async () => {
	const container = document.createElement('div')

	ReactDOM.render(
		<ThemeProvider theme={theme}>
			<IconInfo text="testText" />
		</ThemeProvider>,
		container,
	)
	expect(container.textContent).toMatch('testText')
})
