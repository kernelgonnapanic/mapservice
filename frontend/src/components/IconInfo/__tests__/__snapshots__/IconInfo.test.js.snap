// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render IconInfo component 1`] = `
<DocumentFragment>
  <div
    class="sc-fzplWN cbZktL"
  >
    <div
      class="sc-fzpans ePyOMY"
    />
    <span
      class="sc-fzoLsD cyasPW"
    >
      Hello!
    </span>
  </div>
</DocumentFragment>
`;
